### PR TITLE
Labstarter fix messaging emulator

### PR DESCRIPTION
**Pull request by felipmiguel**
Fix error popup on messaging emulator
Fix UI to have the same styles as main spring petclinic app

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
         <module>spring-petclinic-config-server</module>
         <module>spring-petclinic-discovery-server</module>
         <module>spring-petclinic-api-gateway</module>
-        <!-- <module>spring-petclinic-messaging-emulator</module> -->
+        <module>spring-petclinic-messaging-emulator</module>
```
Comment (vermegi) [inline pom.xml:26]: keep this one commented out please in the labstarter. The lab starts without building the emulator and then builds it separately later in the lab.

File: pom.xml [hunk 2]
```diff
@@ -32,6 +32,7 @@
 
         <spring-boot.version>2.7.6</spring-boot.version>
         <spring-cloud.version>2021.0.4</spring-cloud.version>
+        <version.spring.cloud.azure>4.4.1</version.spring.cloud.azure>
```
Comment (vermegi) [inline pom.xml:35]: Also remove this one again please, it is added later in the lab.

File: pom.xml [hunk 3]
```diff
@@ -61,6 +62,13 @@
                 <version>${assertj.version}</version>
                 <scope>test</scope>
             </dependency>
+            <dependency>
```
Comment (vermegi) [inline pom.xml:65]: Same, remove full dependency, will be added later in the lab.

File: spring-petclinic-customers-service/pom.xml [hunk 4]
```diff
@@ -49,21 +49,13 @@
...
-        </dependency>
 
         <!-- Third parties -->
         <dependency>
-            <groupId>mysql</groupId>
-            <artifactId>mysql-connector-java</artifactId>
-            <scope>runtime</scope>
-        </dependency>
+			<groupId>com.mysql</groupId>
+			<artifactId>mysql-connector-j</artifactId>
```
Comment (vermegi) [inline spring-petclinic-customers-service/pom.xml:56]: Is this indeed "mysql-connector-j" and not "mysql-connector-java"?

File: spring-petclinic-messaging-emulator/src/main/resources/application.yml [hunk 5]
```diff
@@ -1,5 +1,11 @@
-spring:
-  application:
-    name: messaging-emulator
-  # config:
-  #   import: optional:configserver:${CONFIG_SERVER_URL:http://localhost:8888/}
+spring:
+  application:
+    name: messaging-emulator
+  config:
```
Comment (vermegi) [inline spring-petclinic-messaging-emulator/src/main/resources/application.yml:4]: config import should be commented out

File: spring-petclinic-messaging-emulator/src/main/resources/application.yml [hunk 6]
```diff
@@ -1,5 +1,11 @@
...
-  application:
-    name: messaging-emulator
-  # config:
-  #   import: optional:configserver:${CONFIG_SERVER_URL:http://localhost:8888/}
+spring:
+  application:
+    name: messaging-emulator
+  config:
+    import: optional:configserver:${CONFIG_SERVER_URL:http://localhost:8888/}
+  sql:
```
Comment (vermegi) [inline spring-petclinic-messaging-emulator/src/main/resources/application.yml:6]: this is taken from application.yml in the config repo

File: spring-petclinic-messaging-emulator/src/main/resources/application.yml [hunk 7]
```diff
@@ -1,5 +1,11 @@
...
+    init:
+      mode: always
+      platform: hsqldb
+server:
```
Comment (vermegi) [inline spring-petclinic-messaging-emulator/src/main/resources/application.yml:10]: is this needed? It wasn't there in the previous version.

File: spring-petclinic-vets-service/pom.xml [hunk 8]
```diff
@@ -90,8 +82,8 @@
 			<scope>runtime</scope>
 		</dependency>
 		<dependency>
-			<groupId>mysql</groupId>
-			<artifactId>mysql-connector-java</artifactId>
+			<groupId>com.mysql</groupId>
+			<artifactId>mysql-connector-j</artifactId>
```
Comment (vermegi) [inline spring-petclinic-vets-service/pom.xml:86]: Same comment as above: is this correct?

File: spring-petclinic-visits-service/pom.xml [hunk 9]
```diff
@@ -73,8 +65,8 @@
             <artifactId>jolokia-core</artifactId>
         </dependency>
 		<dependency>
-			<groupId>mysql</groupId>
-			<artifactId>mysql-connector-java</artifactId>
+			<groupId>com.mysql</groupId>
+			<artifactId>mysql-connector-j</artifactId>
```
Comment (vermegi) [inline spring-petclinic-visits-service/pom.xml:69]: Same comment as above: is this correct?

**Review by vermegi — CHANGES_REQUESTED** [review]

File: spring-petclinic-customers-service/pom.xml (hunk 4 above)
Comment (felipmiguel) [reply spring-petclinic-customers-service/pom.xml:56]: yes, there is a warning during compile time stating that that is the new version.

File: spring-petclinic-messaging-emulator/src/main/resources/application.yml (hunk 6 above)
Comment (felipmiguel) [reply spring-petclinic-messaging-emulator/src/main/resources/application.yml:6]: how can it take it if not yet configured the config server? the other services have the same config, see this for instance https://github.com/Azure-Samples/spring-petclinic-microservices/blob/labstarter/spring-petclinic-vets-service/src/main/resources/application.yml

File: spring-petclinic-messaging-emulator/src/main/resources/application.yml (hunk 7 above)
Comment (felipmiguel) [reply spring-petclinic-messaging-emulator/src/main/resources/application.yml:10]: if you want to test locally yes. when deployed using config repo can be changed to mysql

**Comment by vermegi** [comment]
LGTM

**Pull request merged**